### DISCUSSION TITLE
Obtaining data for all OSS on OSS Compass

### DISCUSSION THREAD

**Discussion by liaochris** [oss-compass/community#51] [General]
Hi! My name is Chris, and I'm working on a honors research thesis about open source software. The metrics that are provided on OSS Compass would be extremely helpful for me to use in my analysis, but currently, there is not an efficient way to use time series data for all software repositories on OSS Compass for my analysis because I can only analyze one software at a time on the OSS Compass website. 

**I'm asking if it would be possible to obtain a dataset, for each software repository on OSS Compass, of**
1. Repository id + category (from the explore page https://compass.gitee.com/explore)
2. Time series statistics, since inception, for each software on OSS compass in the Productivity, Robustness and Niche Creation categories
3. Inception date for each OSS software

My goal in this project is to understand how potential open source software contributors choose OSS to contribute to, and how their contributions to the OSS codebase or community affect overall usage of that OSS. The data on OSS Compass would be invaluable to me in my research because they are factors that affect the choice contributors make, the degree to which their contributions will be effective and the effect of their contributions. 

This goes without saying, but OSS Compass and any other collaborating organizations (I know CHAOSS data also contributes to OSS Compass's statistics) will of course be acknowledged for their contribution to my thesis.

**Comment by eyehwan** [comment]
Hi @liaochris ,ThankThank you for using OSS-Compass, we have always been happy to cooperate with the academic community！
May I know what kind data do you want? Which repos are you interested on?  For The detailed info, you can contact with me through the mail: yehui.wang.mdh@gmail.com

BR//Yehui.

**Reply by liaochris** [reply]
Hi! I sent you an email awhile back. Let me know how else I can be of service.

**Comment by eyehwan** [comment]
Hi Chris,

I just read your email, sorry for the late reply.

Yes, we are very happy to set up a collaboration with you. But currently we
can't provide you data as a complete dataset to you. But we could provide a
set of APIs to you. Does this work for you?

BR//Yehui.

**Comment by liaochris** [comment]
Dear Yehui,

Yes, that works! What information do you need from me in order to provide
the APIs?

Best,

Chris

**Comment by eyehwan** [comment]
@JuliaZhou2022 As we discussed in the board meeting in the last week,  we have agreed to setup a process to support such acedemic research collaboration. Could you please create an issue to follow this progress? Let @liaochris 's requirement as our first case.

**Comment by JuliaZhou2022** [comment]
Sorry for the late reply! Really busy at the end of the year! We are still working on the OSS Compass data obtaining process. Will proceed it and give reply for this issue ASAP.

**Comment by JuliaZhou2022** [comment]
@liaochris Hi, you can apply OSS Compass data usage now. The detailed OSS Compass Data Usage Application Process can be found by clicking the button "Data usage apply" on the Collaboration page of [ OSS Compass website](https://oss-compass.org/) now. Hope this update is still useful for you.

**Comment by liaochris** [comment]
Awesome, thanks for letting me know! I'll get started on the process

**Reply by JuliaZhou2022** [reply]
Please cc me while sending the email: juliazhou2022@gmail.com.
